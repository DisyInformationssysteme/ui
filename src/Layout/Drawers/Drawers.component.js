import React, { PropTypes } from 'react';
import classnames from 'classnames';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import theme from './Drawers.scss';

function Drawers(props) {
	const container = classnames(theme.container, 'tc-layout-drawers');
	const drawerClasses = classnames(theme.drawer, 'tc-layout-drawers-drawer');
	return (
		<div className={container}>
			<ReactCSSTransitionGroup
				transitionName="tc-layout-drawer"
				transitionAppear
				transitionAppearTimeout={1000}
				transitionEnterTimeout={1000}
				transitionLeaveTimeout={1000}
			>
				{props.drawers.map((drawer, key) => (
					<div key={key} className={drawerClasses}>
						{drawer}
					</div>
				))}
			</ReactCSSTransitionGroup>
		</div>
	);
}

Drawers.propTypes = {
	drawers: PropTypes.arrayOf(PropTypes.element),
};
Drawers.defaultProps = {
	drawers: [],
};

export default Drawers;
