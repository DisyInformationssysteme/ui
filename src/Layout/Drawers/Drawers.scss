// Drawers styles

$tc-drawer-bgcolor: #FFF !default;
$tc-drawer-transition-duration: 0.23s !default;
$tc-drawer-transition-easing: cubic-bezier(0.18, 0.89, 0.32, 1.28) !default;

.container {

	.drawer {
		background-color: $tc-drawer-bgcolor;
		position: fixed;
		top: $navbar-height;
		transform: translateX(0%);
		right:0;
		bottom: 0;
		box-shadow: -1px 2px 5px 1px rgba(0, 0, 0, 0.7);
	}

	:global(.tc-layout-drawer-enter),
	:global(.tc-layout-drawer-appear),
	:global(.tc-layout-drawer-leave) {
		transition: transform $tc-drawer-transition-duration $tc-drawer-transition-easing;
	}

	:global(.tc-layout-drawer-enter),
	:global(.tc-layout-drawer-appear) {
		transform: translateX(100%);

		&:global(.tc-layout-drawer-enter-active),
		&:global(.tc-layout-drawer-appear-active) {
			transform: translateX(0%);
		}
	}

	:global(.tc-layout-drawer-leave) {
		transform: translateX(0%);

		&:global(.tc-layout-drawer-leave-active) {
			transform: translateX(100%);
		}
	}
}
