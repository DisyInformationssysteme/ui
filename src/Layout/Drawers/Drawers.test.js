import React from 'react';
import renderer from 'react-test-renderer';

import Drawers from './Drawers.component';

jest.mock('react-addons-css-transition-group');

const drawers = [
	(<div>
		<h1>Hello drawers</h1>
		<p>You should not being able to read this because I&#39;m first</p>
	</div>),
	(<div>
		<h1>Hello drawers</h1>
		<p>The content dictate the width</p>
	</div>),
];

describe('Drawers', () => {
	it('should render', () => {
		const wrapper = renderer.create(
			<Drawers drawers={drawers} />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	it('should render without drawers', () => {
		const wrapper = renderer.create(
			<Drawers />
		).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
});
