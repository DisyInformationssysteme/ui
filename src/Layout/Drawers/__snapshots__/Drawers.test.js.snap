exports[`Drawers should render 1`] = `
<div
  className="tc-layout-drawers">
  <div
    className="react-addons-css-transition-group">
    <div
      className="tc-layout-drawers-drawer">
      <div>
        <h1>
          Hello drawers
        </h1>
        <p>
          You should not being able to read this because I\'m first
        </p>
      </div>
    </div>
    <div
      className="tc-layout-drawers-drawer">
      <div>
        <h1>
          Hello drawers
        </h1>
        <p>
          The content dictate the width
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawers should render without drawers 1`] = `
<div
  className="tc-layout-drawers">
  <div
    className="react-addons-css-transition-group" />
</div>
`;
